tutorials/tts/4_TTS_FastPitch_Finetuning.ipynb

```diff
@@ -0,0 +1,972 @@
+       "      )\n",
+       "      (1): DiscriminatorS(\n",
+       "        (convs): ModuleList(\n",
+       "          (0): Conv1d(1, 128, kernel_size=(15,), stride=(1,), padding=(7,))\n",
+       "          (1): Conv1d(128, 128, kernel_size=(41,), stride=(2,), padding=(20,), groups=4)\n",
+       "          (2): Conv1d(128, 256, kernel_size=(41,), stride=(2,), padding=(20,), groups=16)\n",
+       "          (3): Conv1d(256, 512, kernel_size=(41,), stride=(4,), padding=(20,), groups=16)\n",
+       "          (4): Conv1d(512, 1024, kernel_size=(41,), stride=(4,), padding=(20,), groups=16)\n",
+       "          (5): Conv1d(1024, 1024, kernel_size=(41,), stride=(1,), padding=(20,), groups=16)\n",
+       "          (6): Conv1d(1024, 1024, kernel_size=(5,), stride=(1,), padding=(2,))\n",
+       "        )\n",
+       "        (conv_post): Conv1d(1024, 1, kernel_size=(3,), stride=(1,), padding=(1,))\n",
+       "      )\n",
+       "      (2): DiscriminatorS(\n",
+       "        (convs): ModuleList(\n",
+       "          (0): Conv1d(1, 128, kernel_size=(15,), stride=(1,), padding=(7,))\n",
+       "          (1): Conv1d(128, 128, kernel_size=(41,), stride=(2,), padding=(20,), groups=4)\n",
+       "          (2): Conv1d(128, 256, kernel_size=(41,), stride=(2,), padding=(20,), groups=16)\n",
+       "          (3): Conv1d(256, 512, kernel_size=(41,), stride=(4,), padding=(20,), groups=16)\n",
+       "          (4): Conv1d(512, 1024, kernel_size=(41,), stride=(4,), padding=(20,), groups=16)\n",
+       "          (5): Conv1d(1024, 1024, kernel_size=(41,), stride=(1,), padding=(20,), groups=16)\n",
+       "          (6): Conv1d(1024, 1024, kernel_size=(5,), stride=(1,), padding=(2,))\n",
+       "        )\n",
+       "        (conv_post): Conv1d(1024, 1, kernel_size=(3,), stride=(1,), padding=(1,))\n",
+       "      )\n",
+       "    )\n",
+       "    (meanpools): ModuleList(\n",
+       "      (0): AvgPool1d(kernel_size=(4,), stride=(2,), padding=(2,))\n",
+       "      (1): AvgPool1d(kernel_size=(4,), stride=(2,), padding=(2,))\n",
+       "    )\n",
+       "  )\n",
+       "  (feature_loss): FeatureMatchingLoss()\n",
+       "  (discriminator_loss): DiscriminatorLoss()\n",
+       "  (generator_loss): GeneratorLoss()\n",
+       ")"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from nemo.collections.tts.models import HifiGanModel\n",
+    "from nemo.collections.tts.models import FastPitchModel\n",
+    "\n",
+    "hifigan_ckpt_path = \"/home/pneekhara/PreTrainedModels/HifiGan--val_loss=0.08-epoch=899.ckpt\"\n",
+    "vocoder = HifiGanModel.load_from_checkpoint(hifigan_ckpt_path)\n",
+    "vocoder.eval().cuda()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "0a4c986f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def infer(spec_gen_model, vocoder_model, str_input, speaker = None):\n",
```

Comment:
Please add a short docstring to explain what this function does.

Reply:
added